tests/lib/rules/no-extra-parens.js

```diff
@@ -129,8 +129,14 @@ ruleTester.run("no-extra-parens", rule, {
         "[a, b]",
         "!{a: 0, b: 1}",
 
+        // ExpressionStatement restricted productions
+        "({});",
+        "(function(){});",
+        "(let[a] = b);",
+        { code: "(function*(){});", parserOptions: { ecmaVersion: 6 } },
+        { code: "(class{});", parserOptions: { ecmaVersion: 6 } },
+
         // special cases
-        "(a + b) * (c + d) == e",
```

Comment:
This is only removed because it's not a special case and it's already covered by two separate tests above.